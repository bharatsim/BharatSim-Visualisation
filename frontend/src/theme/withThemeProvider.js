import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
