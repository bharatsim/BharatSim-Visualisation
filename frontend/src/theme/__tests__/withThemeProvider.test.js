import React from 'react';
import { render } from '@testing-library/react';
import { useTheme } from '@material-ui/core';

import withThemeProvider from '../withThemeProvider';

function DummyChildComponent() {
  const theme = useTheme();
  return (
    <div>
      DummyChildComponent
      <span>{JSON.stringify(theme, null, 2)}</span>
    </div>
  );
}

describe('withThemeProvider HOC', () => {
  it('should match snapshot', () => {
    const Component = withThemeProvider(DummyChildComponent);
    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });
});
