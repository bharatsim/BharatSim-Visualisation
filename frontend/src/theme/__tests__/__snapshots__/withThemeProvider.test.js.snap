// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withThemeProvider HOC should match snapshot 1`] = `
<div>
  <div>
    DummyChildComponent
    <span>
      {
  "breakpoints": {
    "keys": [
      "xs",
      "sm",
      "md",
      "lg",
      "xl"
    ],
    "values": {
      "xs": 0,
      "sm": 600,
      "md": 960,
      "lg": 1280,
      "xl": 1920
    }
  },
  "direction": "ltr",
  "mixins": {
    "toolbar": {
      "minHeight": 56,
      "@media (min-width:0px) and (orientation: landscape)": {
        "minHeight": 48
      },
      "@media (min-width:600px)": {
        "minHeight": 64
      }
    }
  },
  "overrides": {
    "MuiButton": {
      "root": {
        "padding": "8px 16px",
        "height": 40,
        "boxShadow": "unset"
      },
      "sizeSmall": {
        "padding": "8px 16px",
        "height": 32
      },
      "text": {
        "boxShadow": "unset",
        "color": "#594ab7",
        "&:hover": {
          "backgroundColor": "rgba(224, 225, 228, 0.7)"
        },
        "&:focus": {
          "backgroundColor": "rgba(224, 225, 228, 0.7)"
        },
        "&$disabled": {
          "color": "rgba(52, 58, 64, 0.38)"
        }
      },
      "containedPrimary": {
        "boxShadow": "unset",
        "color": "#FFFFFF",
        "backgroundColor": "#594ab7",
        "&:focus": {
          "backgroundColor": "rgba(89, 74, 183, 0.8)"
        },
        "&:hover": {
          "backgroundColor": "rgba(89, 74, 183, 0.8)"
        },
        "&$disabled": {
          "color": "rgba(52, 58, 64, 0.38)",
          "backgroundColor": "#e0e1e4"
        }
      },
      "containedSecondary": {
        "boxShadow": "unset",
        "color": "#594ab7",
        "backgroundColor": "#F0EEF9",
        "&:focus": {
          "backgroundColor": "#F0EEF9"
        },
        "&:hover": {
          "backgroundColor": "#F0EEF9"
        },
        "&$disabled": {
          "color": "rgba(52, 58, 64, 0.38)",
          "backgroundColor": "#e0e1e4"
        }
      },
      "outlined": {
        "boxShadow": "unset",
        "border": "1px solid",
        "color": "#594ab7",
        "borderColor": "#694ED6",
        "&: focus": {
          "backgroundColor": "rgba(89, 74, 183, 0.8)"
        },
        "&: hover": {
          "backgroundColor": "rgba(89, 74, 183, 0.8)"
        }
      }
    },
    "MuiTab": {
      "root": {
        "padding": "4px 8px",
        "minHeight": "unset",
        "minWidth": "unset",
        "height": 32,
        "boxSizing": "border-box",
        "fontFamily": "\\"Roboto\\", sans-serif",
        "fontWeight": 500,
        "fontSize": "0.875rem",
        "lineHeight": 1.2,
        "letterSpacing": "0.01071em",
        "@media (min-width:600px)": {
          "minWidth": "unset"
        }
      },
      "textColorInherit": {
        "color": "rgba(52, 58, 64, 0.64)",
        "borderBottom": "2px solid transparent",
        "opacity": "unset",
        "&$selected": {
          "opacity": "unset",
          "color": "#594ab7",
          "borderBottom": "2px solid",
          "borderColor": "#282154"
        }
      }
    },
    "MuiTabs": {
      "root": {
        "minHeight": 32
      },
      "indicator": {
        "backgroundColor": "transparent"
      }
    },
    "MuiGrid": {
      "root": {
        "margin": 0
      }
    },
    "MuiStepper": {
      "root": {
        "backgroundColor": "transparent",
        "padding": 0
      }
    },
    "MuiStepIcon": {
      "root": {
        "display": "block",
        "color": "rgba(52, 58, 64, 0.38)",
        "&$completed": {
          "color": "#594ab7"
        },
        "&$active": {
          "color": "#594ab7"
        }
      }
    },
    "MuiLink": {
      "root": {
        "color": "#594ab7"
      }
    },
    "MuiDialog": {
      "paperWidthSm": {
        "maxWidth": "unset"
      }
    },
    "MuiDialogContent": {
      "dividers": {
        "padding": 16
      }
    },
    "MuiToolbar": {
      "regular": {
        "minHeight": "unset !important"
      }
    },
    "MuiSlider": {
      "colorPrimary": {
        "color": "#594ab7"
      },
      "track": {
        "height": 4,
        "borderRadius": 0,
        "backgroundColor": "transparent"
      },
      "rail": {
        "height": 4,
        "borderRadius": 0,
        "backgroundColor": "#D2CDEB"
      },
      "mark": {
        "height": 8,
        "width": 8,
        "borderRadius": 4,
        "backgroundColor": "#c5c7cb",
        "marginTop": -2,
        "marginLeft": -4
      },
      "markActive": {
        "backgroundColor": "#c5c7cb",
        "opacity": 1
      },
      "thumb": {
        "height": 12,
        "width": 12,
        "marginTop": -4,
        "marginLeft": -6
      },
      "marked": {
        "marginBottom": 8,
        "marginTop": 4
      },
      "markLabel": {
        "fontFamily": "\\"Roboto\\", sans-serif",
        "fontWeight": 400,
        "fontSize": "0.75rem",
        "lineHeight": 1,
        "letterSpacing": "0.03333em",
        "color": "rgba(52, 58, 64, 0.64)",
        "top": 24
      },
      "valueLabel": {
        "fontFamily": "\\"Roboto\\", sans-serif",
        "fontWeight": 500,
        "fontSize": "0.875rem",
        "lineHeight": 1.57,
        "letterSpacing": "0.00714em",
        "color": "#594ab7",
        "left": "calc(-50% + 12px)",
        "top": -22,
        "& *": {
          "background": "transparent",
          "color": "#594ab7"
        }
      }
    },
    "MuiChip": {
      "outlinedSecondary": {
        "border": "1px solid #493C97"
      }
    },
    "MuiRadio": {
      "colorSecondary": {
        "&$checked": {
          "color": "#594ab7",
          "&:hover": {
            "backgroundColor": "rgba(89, 74, 183, 0.04)",
            "@media (hover: none)": {
              "backgroundColor": "transparent"
            }
          }
        },
        "&$disabled": {
          "color": "rgba(0, 0, 0, 0.26)"
        }
      }
    }
  },
  "palette": {
    "common": {
      "black": "#000",
      "white": "#fff"
    },
    "type": "light",
    "primary": {
      "main": "#282154",
      "light": "#413288",
      "dark": "#535DA8",
      "contrastText": "#fff"
    },
    "secondary": {
      "main": "#494949",
      "light": "#757575",
      "dark": "#A4A4A4",
      "contrastText": "#fff"
    },
    "error": {
      "dark": "#C31E0E",
      "main": "#FD583A",
      "light": "#FF8B66",
      "contrastText": "#fff"
    },
    "warning": {
      "light": "#FFB74D",
      "main": "#FF9800",
      "dark": "#F57C00",
      "contrastText": "rgba(0, 0, 0, 0.87)"
    },
    "info": {
      "light": "#0F74D5",
      "main": "#2593FC",
      "dark": "#70D2FF",
      "contrastText": "#fff"
    },
    "success": {
      "light": "#9EED7E",
      "main": "#57C22D",
      "dark": "#579B21",
      "contrastText": "rgba(0, 0, 0, 0.87)"
    },
    "grey": {
      "50": "#fbfcfe",
      "100": "#e0e1e4",
      "200": "#c5c7cb",
      "300": "#abaeb2",
      "400": "#929599",
      "500": "#797d82",
      "600": "#61666b",
      "700": "#4a4f55",
      "800": "#343A40",
      "900": "#212121",
      "A100": "#d5d5d5",
      "A200": "#aaaaaa",
      "A400": "#303030",
      "A700": "#616161"
    },
    "contrastThreshold": 3,
    "tonalOffset": 0.2,
    "text": {
      "primary": "#343A40",
      "secondary": "rgba(52, 58, 64, 0.64)",
      "disabled": "rgba(52, 58, 64, 0.38)",
      "hint": "rgba(0, 0, 0, 0.38)"
    },
    "divider": "rgba(0, 0, 0, 0.12)",
    "background": {
      "paper": "#fff",
      "default": "#fafafa"
    },
    "action": {
      "active": "rgba(0, 0, 0, 0.54)",
      "hover": "rgba(0, 0, 0, 0.04)",
      "hoverOpacity": 0.04,
      "selected": "rgba(0, 0, 0, 0.08)",
      "selectedOpacity": 0.08,
      "disabled": "rgba(0, 0, 0, 0.26)",
      "disabledBackground": "rgba(0, 0, 0, 0.12)",
      "disabledOpacity": 0.38,
      "focus": "rgba(0, 0, 0, 0.12)",
      "focusOpacity": 0.12,
      "activatedOpacity": 0.12
    }
  },
  "props": {},
  "shadows": [
    "none",
    "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
    "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
    "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
    "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
    "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
    "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
    "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
    "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
    "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
    "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
    "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
    "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
    "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
    "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
    "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
    "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
    "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
    "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)"
  ],
  "typography": {
    "htmlFontSize": 16,
    "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
    "fontSize": 14,
    "fontWeightLight": 300,
    "fontWeightRegular": 400,
    "fontWeightMedium": 500,
    "fontWeightBold": 700,
    "h1": {
      "fontFamily": "\\"Rubik\\", sans-serif",
      "fontWeight": 300,
      "fontSize": "6rem",
      "lineHeight": 1.167,
      "letterSpacing": "-0.01562em"
    },
    "h2": {
      "fontFamily": "\\"Rubik\\", sans-serif",
      "fontWeight": 300,
      "fontSize": "3.75rem",
      "lineHeight": 1.2,
      "letterSpacing": "-0.00833em"
    },
    "h3": {
      "fontFamily": "\\"Rubik\\", sans-serif",
      "fontWeight": 400,
      "fontSize": "3rem",
      "lineHeight": 1.167,
      "letterSpacing": "0em"
    },
    "h4": {
      "fontFamily": "\\"Rubik\\", sans-serif",
      "fontWeight": 400,
      "fontSize": "2.125rem",
      "lineHeight": 1.235,
      "letterSpacing": "0.00735em"
    },
    "h5": {
      "fontFamily": "\\"Rubik\\", sans-serif",
      "fontWeight": 400,
      "fontSize": "1.5rem",
      "lineHeight": 1.334,
      "letterSpacing": "0em"
    },
    "h6": {
      "fontFamily": "\\"Rubik\\", sans-serif",
      "fontWeight": 500,
      "fontSize": "1.25rem",
      "lineHeight": 1.6,
      "letterSpacing": "0.0075em"
    },
    "subtitle1": {
      "fontFamily": "\\"Rubik\\", sans-serif",
      "fontWeight": 400,
      "fontSize": "1rem",
      "lineHeight": 1.75,
      "letterSpacing": "0.00938em"
    },
    "subtitle2": {
      "fontFamily": "\\"Roboto\\", sans-serif",
      "fontWeight": 500,
      "fontSize": "0.875rem",
      "lineHeight": 1.57,
      "letterSpacing": "0.00714em"
    },
    "body1": {
      "fontFamily": "\\"Roboto\\", sans-serif",
      "fontWeight": 400,
      "fontSize": "1rem",
      "lineHeight": 1.5,
      "letterSpacing": "0.00938em"
    },
    "body2": {
      "fontFamily": "\\"Roboto\\", sans-serif",
      "fontWeight": 400,
      "fontSize": "0.875rem",
      "lineHeight": 1.2,
      "letterSpacing": "0.01071em"
    },
    "button": {
      "fontFamily": "\\"Roboto\\", sans-serif",
      "fontWeight": 500,
      "fontSize": "0.875rem",
      "lineHeight": 1.75,
      "letterSpacing": "0.02857em",
      "textTransform": "inherit"
    },
    "caption": {
      "fontFamily": "\\"Roboto\\", sans-serif",
      "fontWeight": 400,
      "fontSize": "0.75rem",
      "lineHeight": 1.66,
      "letterSpacing": "0.03333em"
    },
    "overline": {
      "fontFamily": "\\"Roboto\\", sans-serif",
      "fontWeight": 400,
      "fontSize": "0.75rem",
      "lineHeight": 2.66,
      "letterSpacing": "0.08333em",
      "textTransform": "uppercase"
    }
  },
  "shape": {
    "borderRadius": 4
  },
  "transitions": {
    "easing": {
      "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
      "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
      "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
      "sharp": "cubic-bezier(0.4, 0, 0.6, 1)"
    },
    "duration": {
      "shortest": 150,
      "shorter": 200,
      "short": 250,
      "standard": 300,
      "complex": 375,
      "enteringScreen": 225,
      "leavingScreen": 195
    }
  },
  "zIndex": {
    "mobileStepper": 1000,
    "speedDial": 1050,
    "appBar": 1100,
    "drawer": 1200,
    "modal": 1300,
    "snackbar": 1400,
    "tooltip": 1500
  },
  "colors": {
    "primaryColorScale": {
      "50": "#F0EEF9",
      "100": "#D2CDEB",
      "200": "#B4ABDF",
      "400": "#7769C5",
      "500": "#594ab7",
      "600": "#493C97",
      "700": "#392F75",
      "800": "#282154"
    },
    "grayScale": {
      "50": "#fbfcfe",
      "100": "#e0e1e4",
      "200": "#c5c7cb",
      "300": "#abaeb2",
      "400": "#929599",
      "500": "#797d82",
      "600": "#61666b",
      "700": "#4a4f55",
      "800": "#343A40"
    },
    "button": {
      "borderColor": "#694ED6",
      "color": "#FFFFFF"
    },
    "tab": {
      "color": "#694ED6"
    },
    "borderColor": "#694ED6",
    "textLight": {
      "primary": "#FFFFFF",
      "secondary": "rgba(255, 255, 255, 0.8)",
      "disabled": "rgba(255, 255, 255, 0.6)"
    }
  }
}
    </span>
  </div>
</div>
`;
